code/modules/modular_computers/terminal/terminal.dm

```diff
@@ -33,6 +54,24 @@
 	return TRUE
 
 /datum/terminal/Process()
+	if(current_move)
+		var/result = current_move.update_progress()
+		var/error
```

Comment:
You don't need to declare/set this, just call `append_to_history()` in the if/else.